resources/definitions/fdmprinter.def.json

```diff
@@ -6673,15 +6673,54 @@
                 "prime_tower_brim_enable":
                 {
-                    "label": "Prime Tower Brim",
-                    "description": "Prime-towers might need the extra adhesion afforded by a brim even if the model doesn't. Presently can't be used with the 'Raft' adhesion-type.",
+                    "label": "Prime Tower Base",
+                    "description": "Prime-towers might need the extra adhesion afforded by a brim or raft, even if the model doesn't.",
                     "type": "bool",
-                    "enabled": "resolveOrValue('prime_tower_enable') and (resolveOrValue('adhesion_type') != 'raft')",
-                    "resolve": "resolveOrValue('prime_tower_enable') and (resolveOrValue('adhesion_type') in ('none', 'skirt', 'brim'))",
+                    "enabled": "resolveOrValue('prime_tower_enable') and resolveOrValue('adhesion_type') != 'raft'",
                     "default_value": false,
                     "settable_per_mesh": false,
                     "settable_per_extruder": false
                 },
+                "prime_tower_base_size":
+                {
+                    "value": "resolveOrValue('raft_margin') if resolveOrValue('adhesion_type') == 'raft' else resolveOrValue('brim_width')",
+                    "label": "Prime Tower Base Size",
+                    "description": "The width of the prime tower base.",
+                    "type": "float",
+                    "unit": "mm",
+                    "enabled": "resolveOrValue('prime_tower_enable') and (resolveOrValue('prime_tower_brim_enable') or resolveOrValue('adhesion_type') == 'raft')",
+                    "default_value": "resolveOrValue('brim_width')",
+                    "minimum_value": "0",
+                    "maximum_value": "min(0.5 * machine_width, 0.5 * machine_depth)",
+                    "settable_per_mesh": false,
+                    "settable_per_extruder": false
+                },
+                "prime_tower_base_height":
+                {
+                    "value": "resolveOrValue('layer_height')",
+                    "label": "Prime Tower Base Height",
+                    "description": "The height of the prime tower base.",
+                    "type": "float",
+                    "unit": "mm",
+                    "enabled": "resolveOrValue('prime_tower_enable') and (resolveOrValue('prime_tower_brim_enable') or resolveOrValue('adhesion_type') == 'raft')",
+                    "default_value": 0,
+                    "minimum_value": "0",
+                    "maximum_value": "machine_height",
+                    "settable_per_mesh": false,
+                    "settable_per_extruder": false
+                },
+                "prime_tower_base_curve_magnitude":
+                {
+                    "label": "Prime Tower Base Curve Magnitude",
+                    "description": "The magnitude factor used for the curve of the prime tower foot.",
+                    "type": "int",
```

Comment:
why not use float here? As long as the value is `>0` then it should be alright for non-interger values, no?

Reply:
The main reason is beacause I can't mentally represent a decimal exponant :laughing: so I don't usually use them. But since `std::pow` can handle them, it would work indeed. May be useful for intermediate values between 0 and 2.